src/main/java/org/openrewrite/maven/MavenLoggingResolutionEventListener.java

```diff
@@ -0,0 +1,124 @@
+import org.openrewrite.maven.tree.Scope;
+
+import java.util.List;
+import java.util.Objects;
+
+public class MavenLoggingResolutionEventListener implements ResolutionEventListener {
+
+    private final Log logger;
+
+    public MavenLoggingResolutionEventListener(Log logger) {
+        this.logger = Objects.requireNonNull(logger, "logger cannot be null");
+    }
+
+    @Override
+    public void downloadMetadata(GroupArtifactVersion gav) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Downloading metadata for " + gav);
+        }
+    }
+
+    @Override
+    public void download(GroupArtifactVersion gav) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Downloading " + gav);
+        }
+    }
+
+    @Override
+    public void downloadSuccess(ResolvedGroupArtifactVersion gav, @Nullable ResolvedPom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Downloaded " + gav + pomContaining(containing));
+        }
+    }
+
+    @Override
+    public void downloadError(GroupArtifactVersion gav, List<String> attemptedUris, @Nullable Pom containing) {
+        StringBuilder sb = new StringBuilder("Failed to download " + gav + pomContaining(containing) + "\n");
+        sb.append("Attempted URIs:").append("\n");
+        attemptedUris.forEach(uri -> sb.append("\n  - ").append(uri));
+        logger.error(sb);
+    }
+
+    @Override
+    public void parent(Pom parent, Pom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Resolved parent pom " + containing + pomContaining(containing));
+        }
+    }
+
+    @Override
+    public void dependency(Scope scope, ResolvedDependency resolvedDependency, ResolvedPom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Resolved dependency " + resolvedDependency + " with scope " + scope + pomContaining(containing));
+        }
+    }
+
+    @Override
+    public void bomImport(ResolvedGroupArtifactVersion gav, Pom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Resolved imported dependency " + gav + pomContaining(containing));
+        }
+    }
+
+    @Override
+    public void property(String key, String value, Pom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Resolved property \"" + key + "\" with value \"" + value + "\"" + pomContaining(containing));
+        }
+    }
+
+    @Override
+    public void dependencyManagement(ManagedDependency dependencyManagement, Pom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Resolved managed dependency " + dependencyManagement + " on " + pomContaining(containing));
+        }
+    }
+
+    @Override
+    public void repository(MavenRepository mavenRepository, @Nullable ResolvedPom containing) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Resolving maven repository " + mavenRepository + pomContaining(containing));
```

Comment:
As compared to the others here we use `Resolving` instead of `Resolved`; was that intentional?

Reply:
It is. `resolved` is used twice in the code, both times before the repository resolution happen (https://github.com/openrewrite/rewrite/blob/main/rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java#L230 and https://github.com/openrewrite/rewrite/blob/main/rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java#L508).